src/providers/cryptoauthlib/mod.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+//! Microchip CryptoAuthentication Library provider
+//!
+//! This provider is a hardware based implementation of PSA Crypto, Mbed Crypto.
+use super::Provide;
+use crate::key_info_managers::ManageKeyInfo;
+use derivative::Derivative;
+use log::trace;
+use std::collections::HashSet;
+use std::io::{Error, ErrorKind};
+use std::sync::{Arc, RwLock};
+use uuid::Uuid;
+
+use parsec_interface::operations::list_providers::ProviderInfo;
+use parsec_interface::requests::{Opcode, ProviderID, ResponseStatus, Result};
+
+const SUPPORTED_OPCODES: [Opcode; 0] = [];
+
+/// CryptoAuthLib provider structure
+#[derive(Derivative)]
+#[derivative(Debug, Clone, Copy)]
+pub struct Provider {
+    // device: rust_cryptoauthlib::AtcaDevice,
+}
+
+impl Provider {
+    /// Creates and initialise a new instance of CryptoAuthLibProvider
+    fn new(_key_info_store: Arc<RwLock<dyn ManageKeyInfo + Send + Sync>>) -> Option<Provider> {
+        Some(Provider {})
+    }
+}
+
+impl Provide for Provider {
```

Comment:
It looks like the `list_keys` test is failing because it attempts to list the keys for all providers in the service - thus, including yours because you've added it in the configuration - and that tries to call `list_keys` on this provider, which fails because it uses the default. I think you can just implement it to always return an empty list for now

Reply:
Thank you for the hint. I can reproduce this (it just failed for me the same way) so I can proceed.